### DISCUSSION TITLE
Remove module from build

### DISCUSSION THREAD

**Discussion by Nano-HK** [zephyrproject-rtos/zephyr#48412] [Q&A]
Hi, 

how can I in Zephyr remove a module from build based on conditions? In west.yml I have included modules that are specific for a given board, e.g. nRF. But, those modules and their library, Kconfig and CMakeLists.txt are being included in every build for every board.

Is it possible to exclude a module during build? 

Regards,
Nano

**Comment by yashi** [comment]
Use `add_subdirectory_ifdef(CONFIG_YOUR_BOARD yourdir)`.
ex) https://github.com/zephyrproject-rtos/zephyr/blob/main/modules/hal_nxp/CMakeLists.txt

**Reply by Nano-HK** [reply]
Hello! Thank you for ur reply. 

That is true for local modules or those not defined in west.yml? 

The module is defined in `ZEPHYR_MODULES` variable - in west.yml and is included for every build. What I have done so far is to remove it from `ZEPHYR_MODULES` respective `ZEPHYR_<module>_KCONFIG`  global variable, is that the way to go? Or is there possible to do it in another way?

**Reply by yashi** [reply]
I don't follow what you wrote.

To make conversation clear, I've created a simplest Zephyr module, which doesn't build the source by default.  To build it, you need to select the module in menuconfig.

```
git clone https://github.com/yashi/simplest /tmp/simplest
west build -b qemu_cortex_m3 zephyr/samples/hello_world -- -DZEPHYR_EXTRA_MODULES=/tmp/simplest -DCONFIG_SIMPLEST=y
```
